src/adapter/src/coord/message_handler.rs

```diff
@@ -99,7 +99,7 @@ impl<S: Append + 'static> Coordinator<S> {
         let internal_cmd_tx = self.internal_cmd_tx.clone();
         let client = self.storage_usage_client.clone();
         // Similar to audit events, use the oracle ts so this is guaranteed to increase.
-        let collection_timestamp: EpochMillis = self.get_local_write_ts().await.timestamp.into();
+        let collection_timestamp: EpochMillis = self.peek_local_write_ts().into();
```

Comment:
Is this wrong because it's theoretically possible for two calls here to produce the same timestamps, but we want to guarantee it always increases?

Reply:
Oh I missed this comment, see my other comment. I think it might be wrong.